### DISCUSSION TITLE
Always run in setup mode

### DISCUSSION THREAD

**Discussion by Benji1992-HU** [RavenSystem/esp-homekit-devices#1779] [General]
Hello Everyone!
I have a Shelly RGBW2 device which is earlier is worked correctly with homekit.
Today I would like to update the firmware. 
I flashed the fullhaaboot.bin file like this:
`esptool.py --baud 115200 write_flash -fs 1MB -fm dout -ff 40m 0x0 fullhaaboot.bin`
I manually removed the device from Apple Home App.
After the flashing is finished the device started in setup mode, and I fill up all settings.
The JSON what I used:
`{"c":{"l":2,"b":[{"g":13,"t":5}]},"a":[{"t":30,"r":15,"g":12,"v":14,"w":4,"b":[{"g":5,"p":0},{"g":5,"t":0},{"g":13}]}]}`

When the device rebooted the setup mode started again.
This is my setup screen:
![image0](https://user-images.githubusercontent.com/47895802/157074400-e17bcfd9-2ca5-4420-a9af-5efd643fa9be.png)
When I tap the save button, the page say OK.
Here is the serial monitor log after I tap the save button:
[cng9Kyjd.txt](https://github.com/RavenSystem/esp-homekit-devices/files/8199275/cng9Kyjd.txt)

After this point the device is always started in setup mode and I can't add to Apple Home App. "the device is already added..."

Any suggestion what do I do wrong?

**Comment by jlg89** [comment]
I don't want to assume anything. Have you added a valid WiFi configuration to the HAA config? If not, it will boot to setup, and will not be accessible to HomeKit.

**Reply by pawelosin** [reply]
Try this code {"c":{"l":2,"b":[{"g":13,"t":5}]},"a":[{"t":30,"g":[12,15,14,4],"b":[{"g":5,"p":0},{"g":5,"t":0},{"g":13}]}]
